bftengine/tests/simpleStorage/main.cpp

```diff
@@ -0,0 +1,89 @@
+// Concord
+//
+// Copyright (c) 2018 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0
+// License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#include "FileStorage.hpp"
+#include "../../src/bftengine/assertUtils.hpp"
+
+#include <string.h>
+#include <cassert>
+
+using namespace bftEngine;
+
+struct Object {
+  Object() : elem1(0), elem2(0), elem3(0), elem4(0) {}
+  Object(int e1, long e2, short e3, int e4) :
+      elem1(e1), elem2(e2), elem3(e3), elem4(e4) {}
+
+  int elem1;
+  long elem2;
+  short elem3;
+  int elem4;
+};
+
+const int ELEM1 = 0xaaaaaaaa;
+const long ELEM2 = 0xbbbbbbbbbbbbbbb;
+const short ELEM3 = 0xccc;
+const int ELEM4 = 0xdddddddd;
+
+void verifyData(FileStorage &fileStorage, uint16_t objId) {
+  Object objIn;
+  char *objInBuf = new char[sizeof(objIn)];
+  uint32_t actualObjectSize = 0;
+  fileStorage.read(objId, sizeof(objIn), objInBuf, actualObjectSize);
+  memcpy(&objIn, objInBuf, sizeof(objIn));
```

Comment:
You don't need this memcpy. You can read directly into objIn, by casting it to char* in the call to fileStorage.read. That way you also don't need to allocate objInBuf. Also objInBuf is being leaked here, since it's allocated and not freed.